app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,46 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(benchmark_run_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+
+    category = benchmark_run.result.keys[0]
+
+    previous_benchmark_runs = BenchmarkRun.where('created_at < ?', benchmark_run.created_at).where(initiator_type: Commit, benchmark_type: benchmark_run.benchmark_type,
```

Comment:
Ok I see where your confusion is. Benchmark Runs are not always created in the right order. For example, our server can be running a benchmark_run for commit 2 before it runs the benchmark for commit 1. Therefore, I've reiterated a couple of times that you need to `Find the latest 1000 commits` for that particular benchmark and fetch the benchmark_runs from there.